src/pages/commits/ActiveCommits/buildVariantCard/WaterfallTaskStatusIcon.tsx

```diff
@@ -33,29 +34,45 @@ export const WaterfallTaskStatusIcon: React.VFC<WaterfallTaskStatusIconProps> =
   identifier,
 }) => {
   const { sendEvent } = useProjectHealthAnalytics({ page: "Commit chart" });
+  const [enabled, setEnabled] = useState(false);
   const [loadData, { data, loading }] = useLazyQuery<
     GetFailedTaskStatusIconTooltipQuery,
     GetFailedTaskStatusIconTooltipQueryVariables
   >(GET_FAILED_TASK_STATUS_ICON_TOOLTIP, { variables: { taskId } });
   const { testResults, filteredTestCount } = data?.taskTests ?? {};
   const failedTestDifference = filteredTestCount - testResults?.length;
 
+  let timeout;
   const onHover = () => {
-    // Only query failing test names if the task has failed.
-    if (isFailedTaskStatus(status)) {
-      loadData();
+    timeout = setTimeout(() => {
+      setEnabled(true);
+      // Only query failing test names if the task has failed.
+      if (isFailedTaskStatus(status)) {
+        loadData();
+      }
+    }, 500);
+  };
+  const onMouseLeave = () => {
+    setEnabled(false);
+    if (timeout) {
+      clearTimeout(timeout);
     }
   };
-
+  useEffect(() => () => {
+    if (timeout) {
+      clearTimeout(timeout);
+    }
+  });
```

Comment:
I think this is running on every hover (at least that's what I'm seeing when I add a print statement here). I think we just want this to run when the component unmounts?

Reply:
Nice catch. I shouldn't have listened to eslint and removed the dep array.